cas-server-core-tickets/src/main/java/org/jasig/cas/web/flow/TicketExceptionHandler.java

```diff
@@ -0,0 +1,95 @@
+package org.jasig.cas.web.flow;
+
+import com.google.common.collect.ImmutableList;
+import org.jasig.cas.ticket.AbstractTicketException;
+import org.jasig.cas.ticket.UnsatisfiedAuthenticationPolicyException;
+import org.springframework.binding.message.MessageBuilder;
+import org.springframework.binding.message.MessageContext;
+import org.springframework.stereotype.Component;
+
+import javax.validation.constraints.NotNull;
+import java.util.List;
+
+/**
+ * Performs two important error handling functions on an {@link AbstractTicketException} raised from the login webflow
+ * layer:
+ *
+ * <ol>
+ *     <li>Maps ticket errors onto message bundle strings for display to user for configured errors only.</li>
+ *     <li>Determines the next webflow state by comparing ticket errors against {@link #errors} in list order. The first
+ *     entry that matches determines the outcome state, which is the simple class name of the exception.</li>
+ * </ol>
+ *
+ * @author Daniel Frett
+ * @since 4.3.0
+ */
+@Component("ticketExceptionHandler")
+public class TicketExceptionHandler {
```

Comment:
Alright, let me see about refactoring this to combine with the other handler. There are some differences in how we access the actual exception, for TicketExceptions its direct, for AuthenticationException we actually process exceptions in handler failures.